Core/StitchCoreSDK/Sources/StitchCoreSDK/Auth/Internal/CoreStitchAuth+StitchAuthRequestClient.swift

```diff
@@ -44,6 +44,24 @@ extension CoreStitchAuth {
         }
     }
 
+    public func doAuthenticatedRequestOptionalResult<T: Decodable>(_ stitchReq: StitchAuthRequest) throws -> T? {
+        let response = try self.doAuthenticatedRequest(stitchReq)
+        do {
+            guard let responseBody = response.body,
+                let responseString = String.init(data: responseBody, encoding: .utf8) else {
+                    throw StitchError.serviceError(
+                        withMessage: StitchErrorCodable.genericErrorMessage(withStatusCode: response.statusCode),
+                        withServiceErrorCode: .unknown)
+            }
+
+            do {
+                return try BSONDecoder().decode(T.self, from: responseString)
```

Comment:
[minor] I'm not sure this is quite what you want. I think you just want to get rid of the `do-catch` block and replace it with

```swift
try BSONDecoder().decode(T?.self, from: responseString)
```
If there actually is a decoding error, you want to propagate it.

Though it's possible that the Swift driver doesn't support decoding a BSON undefined/null into a `nil`. In that case, I guess we can leave this as is, but this method will mask all decoding errors by returning a `nil` result. This is probably fine for now, but ideally we want to be able to distinguish `nil` from an invalid response.

Reply:
Sadly, that does not work :(